web3-dev/smart-contracts/smart-contract-example-sum.rst

```diff
@@ -124,61 +84,63 @@ Smart-contract can be compiled using the command:
 
     npm run build
 
+Note that a `build/deployer.wasm` file has also been generated. It will be used to deploy your contract on Massa network.
+
 .. _sending-sc-sum:
 
-Putting your smart-contract on the blockchain
----------------------------------------------
+Deploy your smart-contract on the blockchain
+--------------------------------------------
 
 We'll now turn to the process of putting the smart-contract on the Massa blockchain.
 
 For the deployment, you will need a wallet with some coins. 
 
+To send transaction on the network, you need to provide your wallet private key.
+This is done using environment variable in `.env` file.
+
 .. code-block::
 
     cp .env.example .env
 
-This command will create a `.env` file. Now fill it with your wallet keys and address.
+This command will create a `.env` file. Now fill it with your wallet private key.
 
-Deploying a smart-contract on Massa blockchain is done by calling a temporary smart-contract that will store 
-our sum smart-contract onto the ledger.
+Contract deployment is done by calling a deployer smart-contract which can be found here: `deployer/deployer.as.ts`. It will store our sum smart-contract onto the ledger. The deployer contract already includes your compiled `main.ts` contract and has been itself compiled at the `npm run build` step.
 
-Sending the smart-contract to the Massa blockchain is done with the command `npm run deploy`.
+We will send the deployer smart-contract to the Massa blockchain with:
 
 .. code-block::
 
-    npm run deploy build/main.wasm
+    npm run deploy
 
-This command will execute `main.ts` smart-contract, and this smart-contract will store the `sum.ts` smart-contract onto the ledger.
+This command will execute the compiled deployer `deployer.wasm`, and this smart-contract will store the `main.ts` smart-contract onto the ledger.
 
 You will see an output like this:
 
 .. code-block::
 
-    > my-massa-sc@1.0.0 deploy
-    > ts-node --esm deployer/deployment_script.ts build/main.wasm
+    > my-massa-sc@0.0.1 deploy
+    > ts-node deployer/deploy-contract.ts
 
-    Smartcontract file path : build/main.wasm
+    Deploying smartcontract: build/deployer.wasm
 
-    Deployment has begun...
+    Operation submitted successfully to the network. Operation id: ntUDiV9tJJv94Qi4tNzZHZpCoBGUGAyjejZTuQi4eERAaJPCD
 
-    Deployment successfully ended with operation id CyZxiMa53WyLYjKUYrfEYCyaEvXU5EdUamUhNyGmATrd7M9Tx
+    Waiting for the state of operation to be Final... this may take few seconds
 
-    Retrieving first event...
+    Deployment success with event: Contract deployed at address: A1PjpgXyXSBeiG1rbXCP4ybhVccYzpysDKYmkymXWd81idutaD9
```

Comment:
```suggestion
    Deployment success with event: Contract deployed at address: <smart-contract address>
```

Reply:
no I want to keep this one as is because it is re-used bellow